chore: relax dependency versions

nmoutschen opened the pull request:
Hi! 👋🏻 

We're using a different version of the `tracing-core` crate, which causes issues when we try to add this crate to our project, as the exact version are incompatible. I've tried to relax the tracing crates version requirements and was able to get this working without any problem.

Would it be possible to relax the version here so we don't have to depend on our own fork on this crate?

NAlexPear commented:
My only hesitancy would be bumping lower than `v0.1.22` for `tracing-core` since that's the version where [`valuable` support was added](https://github.com/tokio-rs/tracing/discussions/1906). Since this library officially supports `valuable` as well, it would be easiest on my end to restrict `tracing-core` to `0.1.22` (or `>= 0.1.22 < 0.2.0` in practice). Would that be sufficient for you use-cases?

nmoutschen commented:
Hey @NAlexPear! Thanks for the quick reply here!

We're currently pinning `0.1.26` so there's no problem there. I've set the minimum version for `tracing-core` to `0.1.22`, but kept the other crates at the minor level only. Would that work for you?

NAlexPear commented:
@nmoutschen is there a reason to relax those other dependency versions? If we can get away with just pinning `tracing-core`, then I'd prefer to do that.

I'd also be against `~` here. The [default behavior](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio) of unprefixed dependency specs should be fine.

nmoutschen commented:
Hey @NAlexPear! Apologies for the delay here, I didn't have time to test version combinations until now.

The versions of the other crates were dependent on versions of `tracing-core` newer than `0.1.22`, which would cause issues too. `tracing-0.1.35` depends on `tracing-core-^0.1.27`, and `tracing-subscriber-0.3.15` depends on `tracing-core-^0.1.28`, both versions are newer than `0.1.26`.

I've updated to put the latest version that depends on `0.1.22` and removed the `~`.

NAlexPear commented:
LGTM, thanks for the PR!
